openadapt/evaluators/evaluator.py

```diff
@@ -0,0 +1,204 @@
+
+        Args:
+            point1 (Tuple[float, float]): first point
+            point2 (Tuple[float, float]): second point
+
+        Raises:
+            ValueError: if either point is not 2D
+
+        Returns:
+            float: euclidean distance between the two points
+        """
+        if len(point1) != 2 or len(point2) != 2:
+            raise ValueError("Both points must be 2D coordinates (x, y)")
+
+        x1, y1 = point1
+        x2, y2 = point2
+        distance = math.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)
+        return distance
+
+    def _score_valid_actions(
+        self,
+        ref_action: KeyAction | MouseAction,
+        prediction_action: KeyAction | MouseAction,
+    ) -> float:
+        """Score the validity of the predicted action
+
+        Args:
+            ref_action (KeyAction | MouseAction): reference action
+            prediction_action (KeyAction | MouseAction): predicted action
+
+        Returns:
+            float: score of the predicted action
+        """
+        score = 0
+        # verify if ref_action and prediction_action are of the same object type
+        if type(ref_action) != type(prediction_action):
+            # if the actions are not of the same type, return 0.1, as a correct action
+            # is already worth the score
+            return score + 0.1
+        # if it is same key action, same button pressed, give full score
+        if type(ref_action) == KeyAction:
+            score += 0.5
+            if ref_action.name == prediction_action.name:
+                score += 0.25
+            if ref_action.key_name == prediction_action.key_name:
+                score += 0.25
+            return score
+
+        # for mouse actions, give score for each attribute
+        # the point click is compared by Euclidean distance to see how close the
+        # predicted point is to the reference point
+        if type(ref_action) == MouseAction:
+            score += 0.5
+
+            if ref_action.name == prediction_action.name:
+                score += 0.10
+            if ref_action.mouse_button_name == prediction_action.mouse_button_name:
+                score += 0.10
+            if ref_action.mouse_pressed == prediction_action.mouse_pressed:
+                score += 0.10
+
+            ref_mouse_pos = (ref_action.mouse_x, ref_action.mouse_y)
+            prediction_mouse_pos = (
+                prediction_action.mouse_x,
+                prediction_action.mouse_y,
+            )
+            distance = self._euclidean_distance(ref_mouse_pos, prediction_mouse_pos)
+            normalized_distance = distance / self.max_screen_size[0]
+            # the smaller the distance, the better the score
+            additional_score = 0.1 - normalized_distance
+            score += additional_score
+            return score
+
+    def parse_completion(self, completion: str) -> KeyAction | MouseAction:
+        """Parse a completion string to a KeyAction or MouseAction
```

Comment:
This method, though not sure if it should be tightened per model specific...

I keep it here for now, but likely each model would have to re-implement this method to parse completion to actions.